Libraries/CustomComponents/ListView/ListView.js

```diff
@@ -484,9 +484,8 @@ var ListView = React.createClass({
   },
 
   _getDistanceFromEnd: function(scrollProperties) {
-    return scrollProperties.contentLength -
-      scrollProperties.visibleLength -
-      scrollProperties.offset;
+    var maxHeight = Math.max(scrollProperties.contentLength, scrollProperties.visibleLength);
```

Comment:
In case the `dataSource` is empty, the `contentLength` is potentially less than the `visibleLength`.